app/experimenter/static/js/edit-objectives.js

```diff
@@ -0,0 +1,22 @@
+
+function updateSurvey() {
+  const survey = $('[type="radio"][name="survey"][value="True"]:checked').length > 0;
+
+  // We hide the confusing "Optional" label manually here
+  $('#survey-url-and-instructions .optional-marker').hide();
+
+  if (survey) {
+    $('#survey-url-and-instructions').show()
+  } else {
+      $('#survey-url-and-instructions').hide();
+  }
+}
+
+
+
+$('[type="radio"]').change(function (e) {
+  updateSurvey();
+
+});
+
+updateSurvey();
```

Comment:
I took the 'hiding of the optional button' from the risk-technical section... where we show and hide an additional text box there (Though it's done incorrectly there and I plan to fix it in a separate PR). It makes total sense in the risk-technical section, because if a user says Yes to the risk-technical question, they are required then to put an explanation in the text box. I am debating whether it makes sense here... because urls and instructions are _not_ required by the form in the formal sense... but if you answer Yes to the survey... you really do need to add the URLs and instructions (I'm assuming). So I'm not sure what to do about that.

Reply:
also I used Jquery because risk-technical stuff used it..

Reply:
I think using jQuery for consistency is fine.

I'm a bit torn myself about hiding the `Optional` label. According to Shell's remarks on Slack, we *do* need this information, but we *don't* want to block the form submission. In other words, the fields are really optional. I think we should assume the user will read the lack of `Optional` as indicating the fields are required, which would make (some of) them not submit it, which almost makes the form required. I think we shouldn't get too fancy and just leave the `Optional` labels as they are.

I do wonder if it's possible to do this more simply, but I think it isn't. We can't use CSS trickery to hide/show the following elements because the `:checked` radio will be inside some `div`s. Can we use Bootstrap collapse even though it's a `radio` instead of a button?

Reply:
It might be possible with bootstrap.

Reply:
I removed the optional stuff and just wondering if we think I should explore switching this over to bootstrap  or leave as is.

Reply:
wondering also what @jaredkerim is feeling.

Reply:
@uhlissuh @glasserc Yeah so this is an ugly situation we've backed ourselves into.

Originally the design was that some fields were 'required' in the django form sense, and some were 'optional'.  So then on a given form page, you couldn't save the form at all without all the required fields being completed, and then the optional ones may or may not need to be completed before you can move to 'ready to ship'.

But then people complained that they wanted to save some fields and not others at the same time (because they don't know the answers yet) so we started changing some 'required' fields to 'optional' in the django form sense, even though they're 'delayed required' ie block ready to ship. 

So now we're in this stupid situation where fields say 'optional' even though they're more like 'delayed required' and we don't make that clear anywhere.

So we filed this ticket:  https://github.com/mozilla/experimenter/issues/902 to address this more concretely.  The point of that ticket is to basically move every single field to be 'django optional' so nothing is required to be able to save a form page, but fields that are still 'delayed required' continue to behave that way.  

Part of that change will be to take out those 'optional' labels we show.  So I would say just take out that one line of jquery that hides the optional label, leave the rest as is, and we can fix that more coherently in the other ticket. /rant

Reply:
I took the optional stuff out of the jQuery.